estimagic/estimation/estimate.py

```diff
@@ -0,0 +1,83 @@
+import numpy as np
+
+from estimagic.config import DEFAULT_DATABASE_NAME
+from estimagic.decorators import aggregate_criterion_output
+from estimagic.optimization.optimize import maximize
+from estimagic.optimization.process_arguments import process_optimization_arguments
+
+
+def estimate_likelihood(
+    criterion,
```

Comment:
I think it is confusing to call the first argument of `criterion`. What the user passes here is not what the user would pass to `maximize` if she wanted to do the same thing. I would suggest `criterion_contributions` or `contribution_level_criterion`.

Reply:
I changed it to ``loglikeobs``. Maybe we want to find a more general name in the future when there is a GMM interface.

Reply:
I think it definitely should have a plural in there. And then a clear docstring stating that this is supposed to be an array of log-likelihood contributions (I am mostly guessing here). `criterions_per_obs` ? That should work for GMM & friends, too.

You probably discussed that elsewhere, but is there a reason for `logging` & `log_options` instead of `logging_options`? Not that I would think there should be many options for taking logarithms, but expanding `log` to `logarithm` is just hard-wired in my brain when a likelihood is close by. But maybe I am just getting too old...

Reply:
I think we chose ``log_options`` instead of ``logging_options`` for brevity. But, this is probably not strictly necessary and not a good reason if it causes confusion.

I moved the rest of the discussion to #112 as I would like to have this merged and Janos should join the discussion.